src/FastState.cpp

```diff
@@ -69,51 +69,30 @@ bool FastState::is_move_legal(int color, int vertex) {
                 !board.is_suicide(vertex, color));
 }
 
-void FastState::play_pass(int color) {
-    m_movenum++;
-
-    m_lastmove = FastBoard::PASS;
+void FastState::play_move(int vertex) {
+    play_move(board.m_tomove, vertex);
+}
 
+void FastState::play_move(int color, int vertex) {
     board.m_hash ^= Zobrist::zobrist_ko[m_komove];
-    m_komove = 0;
+    m_komove = board.update_board(color, vertex);
     board.m_hash ^= Zobrist::zobrist_ko[m_komove];
 
+    m_lastmove = vertex;
+    m_movenum++;
+
     if (board.m_tomove == color) {
         board.m_hash ^= 0xABCDABCDABCDABCDULL;
     }
     board.m_tomove = !color;
 
     board.m_hash ^= Zobrist::zobrist_pass[get_passes()];
-    increment_passes();
-    board.m_hash ^= Zobrist::zobrist_pass[get_passes()];
-}
-
-void FastState::play_move(int vertex) {
-    play_move(board.m_tomove, vertex);
-}
-
-void FastState::play_move(int color, int vertex) {
-    if (vertex != FastBoard::PASS) {
-        board.m_hash ^= Zobrist::zobrist_ko[m_komove];
-        m_komove = board.update_board(color, vertex);
-        board.m_hash ^= Zobrist::zobrist_ko[m_komove];
-
-        m_lastmove = vertex;
-        m_movenum++;
-
-        if (board.m_tomove == color) {
-            board.m_hash ^= 0xABCDABCDABCDABCDULL;
-        }
-        board.m_tomove = !color;
-
-        if (get_passes() > 0) {
-            board.m_hash ^= Zobrist::zobrist_pass[get_passes()];
-            set_passes(0);
-            board.m_hash ^= Zobrist::zobrist_pass[0];
-        }
-    } else if (vertex == FastBoard::PASS) {
-        play_pass(color);
+    if (vertex == FastBoard::PASS) {
+        increment_passes();
+    } else {
+        set_passes(0);
     }
+    board.m_hash ^= Zobrist::zobrist_pass[get_passes()];
```

Comment:
When passing,  `0` and the number of passes are xored in.
When not passing, the hash contains no `0` passes modification. 
Would it not be more consistent if `calc_hash()` would also xor in "`0` passes" so that the hash would only contain the right number of passes?

Reply:
I think it would be nice to fix the intercoupled of zobrist logic between FastState and FullBoard, so that calc_hash() == m_hash at all times.

That being said I don't think this PR is the right place for changing it as I'd like to keep it a no-op change.